NumberTheory/PerfectNumbers.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+    Perfect number:
+    Checks if a given number is perfect. Could be used to generate a 
+    sequence of perfect numbers
+    Time complexity:
+    O(N), where N is the number being checked 
+    Space complexity:
+    O(1), constant amount of space needed
+*/
+
+#include <iostream>
+
+using namespace std;
+
+ULL is_perfect(int num) {
+	int sum = 1;
+	for(int divisor = 2; divisor <= num/2; divisor++)
+	{
+		if(num % divisor == 0)
+			sum += divisor;
+	}

```

Comment:
The divisors can be found in _O_(sqrt(N)) by checking only upto sqrt(N), and adding `divisor` and `num / divisor` to `sum`.